framework/web/Request.php

```diff
@@ -195,29 +227,61 @@ public function resolve()
     }
 
     /**
+     * Filters headers according to the trusted host config.
+     * @param array $headers
+     * @since 2.0.12
+     * @return array The filtered list of headers.
+     */
+    protected function filterHeaders(array $headers)
+    {
+        $host = $this->getUserHost();
+        // Don't use getUserIP here since that depends on using headers.
+        $ip = $this->getUserIP(false);
+        foreach($this->trustedHostConfig as $hostRegex => $trustedHeaders) {
+            if (!is_array($trustedHeaders)) {
+                $hostRegex = $trustedHeaders;
+                $trustedHeaders = $this->secureHeaders;
+            }
+
+            if (preg_match($hostRegex, $host) || preg_match($hostRegex, $ip)) {
+                foreach($this->secureHeaders as $secureHeader) {
+                    if (!in_array($secureHeader, $trustedHeaders)) {
+                        unset($headers[$secureHeader]);
+                    }
+                }
+                return $headers;
+            }
+        }
+        return [];
+    }
+
+    /**
      * Returns the header collection.
      * The header collection contains incoming HTTP headers.
      * @return HeaderCollection the header collection
      */
     public function getHeaders()
     {
         if ($this->_headers === null) {
+            // Check hosts.
```

Comment:
This wasn't a separate method before. What would it be called and would it do more than store the result in a private variable?